tests/instruments/test_instrument.py

```diff
@@ -121,30 +120,61 @@ def test_read(self, instr):
         instr.read()
         assert instr.adapter.method_calls == [mock.call.read()]
 
+    def test_write_bytes(self, instr):
+        instr.write_bytes(b"abc")
+        assert instr.adapter.method_calls == [mock.call.write_bytes(b"abc")]
+
+    def test_read_bytes(self, instr):
+        instr.read_bytes(5)
+        assert instr.adapter.method_calls == [mock.call.read_bytes(5)]
+
+
+class TestWaiting:
+    @pytest.fixture()
+    def instr(self):
+        class Faked(Instrument):
+            def wait_till_read(self, query_delay=0):
+                self.waited = query_delay
+        return Faked(ProtocolAdapter(), name="faked")
+
     def test_ask(self, instr):
```

Comment:
With the new change, you're only testing that the instrument did not wait -- difference to test_ask_time? did you mean to delete the assert about the ask call?

Reply:
With `test_waiting` I test, whether the Instrument really waits. The other tests (`ask`, `ask_time`...) test, whether the waiting method has been called. In these tests I do not actually sleep, for the sleeping tests fail sometimes due to excessive calculation and actual waiting prolongs the test execution.

Reply:
I see. Maybe a better test name would be `test_ask_calls_wait`, `test_ask_calls_wait_with_delay` etc.? 
This is useful because if the test fails, it's clearer what specifically failed to test.

Reply:
done